File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/pojo/UsersRobot.java
package cz.muni.fi.umimecesky.pojo;

import android.widget.ImageView;

public class UsersRobot extends AbstractRobot {

    public UsersRobot(ImageView view) {
        super(view, 1);
    }

    @Override
    public void applyCorrect() {
        if (getLastWasCorrect()) super.moveForward();
        setLastWasCorrect(true);
    }

    @Override
    public void applyWrong() {
        moveBackward();
    }
}

File: Roboti/app/src/main/java/cz/muni/fi/umimecesky/pojo/UsersRobot.kt
package cz.muni.fi.umimecesky.pojo

import android.widget.ImageView

class UsersRobot(view: ImageView) : AbstractRobot(view, 1) {

    override fun applyCorrect() {
        if (lastWasCorrect) super.moveForward()
        lastWasCorrect = true
    }

    override fun applyWrong() = moveBackward()
}
